Emake Makefile Server Regression

Something related to fundies changes (9eefc9a850162617781cf86c2575a09f9fca68ad) causes `server.pb.h` not to get generated during the first make run. It has to be run twice in order to successfully build.
```
make[1]: Leaving directory '/c/Users/Owner/Documents/Projects/enigma-dev/shared/libpng-util'
protoc -I. --cpp_out=.eobjs Shader.proto
protoc -I. --cpp_out=.eobjs Object.proto
protoc -I. --cpp_out=.eobjs Room.proto
Shader.proto:4:1: warning: Import options.proto is unused.
protoc -I. --cpp_out=.eobjs compiler.proto
protoc -I. --cpp_out=.eobjs Script.proto
protoc -I. --cpp_out=.eobjs Timeline.proto
Script.proto:4:1: warning: Import options.proto is unused.
protoc -I. --cpp_out=.eobjs treenode.proto
protoc -I. --cpp_out=.eobjs Sprite.proto
protoc -I. --cpp_out=.eobjs Settings.proto
protoc -I. --cpp_out=.eobjs Path.proto
protoc -I. --grpc_out=.eobjs --plugin=protoc-gen-grpc=/mingw64/bin/grpc_cpp_plugin.exe server.proto
make[1]: *** No rule to make target '.eobjs/server.grpc.pb.h', needed by '.eobjs/Action.pb.o'.  Stop.
make[1]: Leaving directory '/c/Users/Owner/Documents/Projects/enigma-dev/shared/protos'
make: *** [Makefile:31: libProtocols] Error 2
```

--- a/Default.mk
+++ b/Default.mk
@@ -41,7 +41,9 @@ $(OBJ_DIR)/shared/%.o: $(SHARED_SRC_DIR)/%.cpp | obj_dirs
 	$(CXX) $(CXXFLAGS) -MMD -c -o $@ $<
 
 # Proto rules
-$(OBJ_DIR)/%.pb.o $(OBJ_DIR)/%.pb.d: | $(GENERATED)
+# The order-only dependency on $(GENERATED) is an over-approximation to force all protos to be built
+# before any object files. This matters on the first run when Make can't yet know about #includes.
+$(OBJ_DIR)/%.pb.o $(OBJ_DIR)/%.pb.d: $(OBJDIR)/%.pb.cc | $(GENERATED)
 	$(CXX) $(CXXFLAGS) -MMD -MP -c -o $(OBJ_DIR)/$*.pb.o $(OBJ_DIR)/$*.pb.cc
 
 $(OBJ_DIR)/%.pb.cc $(OBJ_DIR)/%.pb.h: %.proto | obj_dirs 

